packages/ui/src/views/w3m-wallet-explorer-view/index.ts

```diff
@@ -173,49 +108,53 @@ export class W3mWalletExplorerView extends LitElement {
     else await this.onConnect(listing.desktop, listing.name)
   }
 
+  private readonly searchDebaunce = debounce((value: string) => {
+    this.firstFetch = true
+    this.endReached = false
+    this.search = value
+    ExplorerCtrl.resetSearch()
+    this.fetchWallets()
+  })
+
+  private onSearchChange(event: Event) {
+    const { value } = event.target as HTMLInputElement
+    if (value.length >= 3) this.searchDebaunce(value)
+    else if (this.search) {
+      this.search = undefined
+      this.endReached = this.isLastPage()
+      ExplorerCtrl.resetSearch()
+    }
+  }
+
   // -- render ------------------------------------------------------- //
   protected render() {
-    const { listings } = this.listingResponse
+    const { wallets, search } = ExplorerCtrl.state
+    const { listings } = this.search ? search : wallets
     const classes = {
-      'w3m-loading': this.isLoadingEndless,
-      'w3m-first-fetch': this.firstFetch,
+      'w3m-loading': this.loading && !listings.length,
       'w3m-end-reached': this.endReached
     }
 
     return html`
       ${dynamicStyles()}
 
-      <w3m-modal-header>${this.searchTemplate()}</w3m-modal-header>
+      <w3m-modal-header>
+        <w3m-search-input .onChange=${this.onSearchChange.bind(this)}></w3m-search-input>
+      </w3m-modal-header>
 
       <w3m-modal-content class=${classMap(classes)}>
-        ${this.isLoadingFiltered && !this.isLoadingEndless && !this.firstFetch
-          ? this.loadingTemplate()
-          : html`
-              ${listings.length > 0
-                ? html`
-                    <div class="w3m-content">
-                      ${listings.map(
-                        listing => html`
-                          <w3m-wallet-button
-                            src=${listing.image_url.lg}
-                            name=${listing.name}
-                            .onClick=${async () => this.onConnectPlatform(listing)}
-                          >
-                          </w3m-wallet-button>
-                        `
-                      )}
-                    </div>
-                  `
-                : html`
-                    ${this.firstFetch
-                      ? null
-                      : html`
-                          <div class="w3m-centered-block">
-                            <w3m-text align="center" color="secondary">No results found</w3m-text>
-                          </div>
-                        `}
-                  `}
-            `}
```

Comment:
Sorry if I'm missing something, but don't we need this?

Reply:
Noup, it is handled via https://github.com/WalletConnect/web3modal/pull/597/files#diff-3f421d6903c140850f3c5abc03fb4209e31026cd27ba8ea65e3a92a6da1cfdabR56-R58